array/array.go

```diff
@@ -223,56 +223,146 @@ func (a *String) ValueLen(i int) int {
 }
 
 type StringBuilder struct {
-	b *array.BinaryBuilder
+	b            *array.BinaryBuilder
+	mem          memory.Allocator
```

Comment:
Is `allocator` a better name, or does the interface present itself as if it were memory?

Reply:
Generally I've used `mem` for things that directly use the arrow memory allocator because that's what the parameter is called in the arrow library. I find `alloc` to be a bit weird to read and `allocator` too long.

An example for the arrow library: https://pkg.go.dev/github.com/apache/arrow/go/arrow/array#NewBinaryBuilder.